### DISCUSSION TITLE
How to close jest major update Pull Request

### DISCUSSION THREAD

**Discussion by deraw** [renovatebot/renovate#10710] [Request Help] [answered]
Hello,

[A Pull Request](https://github.com/assurance-maladie-digital/design-system/pull/1213) was created on a repository I'm working on to update jest and ts-jest to v27 and I can't close it.

I've read other discussions where it was resolved by updating the configuration but it seems very odd to have to update my configuration to ignore this particular update, why can't I close the Pull Request like this one https://github.com/assurance-maladie-digital/design-system/pull/1125 that updates ts-jest to v27? It's the same major version bump so this is confusing

Thanks!

**Comment by rarkins** [comment]
I agree this is undesirable but not sure if the solution. Ideally it should say "to v27" so that it's not immortal.

**Reply by herndlm** [reply]
Interesting, I got the exact same problem.

<details>
  <summary>Gitlab react-monorepo MR (OK)</summary>
<img width="640" alt="image" src="https://user-images.githubusercontent.com/5738896/151130183-ffc54dea-29b4-41dd-988e-7e533e4bd928.png">
</details>

<details>
  <summary>Gitlab jest MR (not OK)</summary>
<img width="627" alt="image" src="https://user-images.githubusercontent.com/5738896/151130274-e24966e5-25c8-4393-b6c0-81a7dea23de0.png">
</details>

<details>
  <summary>Bitbucket composer PR (not OK)</summary>
<img width="606" alt="image" src="https://user-images.githubusercontent.com/5738896/151130989-e83ca5c1-ec53-4afc-b098-13215cec48d5.png">
</details>

So did I get it right that the problem behind this is that it doesn't add the version to the title, which I guess is related to renovate enabling `recreateClosed`? Maybe I can try looking into this with my self-hosted setup.

UPDATE: Created https://github.com/renovatebot/renovate/issues/13827 for this

**Comment by karfau** [comment]
I have the same issue with jest v28 PRs which updates jest and ts-jest, but doesn't add the version to the PR title:
![image](https://user-images.githubusercontent.com/135657/168411487-f02f9775-ae90-49fe-9a0f-32259cf64c1c.png)
the new major verison is in the branch name though. So it's not clear why it is not putting it into the  title.
Also the linked ticket has been marked as closed, so maybe it just looks similar, but the problem is a different one?

This is a major issue for us because until this is solved, all repos that reach this point are not receiving any other automated updates.

I even configured `dependencyDashboardApproval` for major updates:
https://github.com/bettermarks/renovate-config/commit/c68aaeca2e832b625be154c7bad82822dd627cea

But even retrying/rebasing the PR or rerunning renovate on the whole repo, the PR stays immortal and I also verified it by closing it.

PS: I decided to open this thread in the existing discussion, since the title was a perfect match. Just let me know in case I should file it somewhere else.

**Reply by rarkins** [reply]
It really looks like the title should have "to v28". Can you get it reproducing in a minimal repo?

**Reply by karfau** [reply]
Here is the "reproduction":
https://github.com/karfau/renovate-10710/issues/2
At least the `dependencyDashboardApproval` is now applied correctly.
But I'm expecting that the PR will be immortal when created, since it will not have a version number.

So beside my questions above, the next one would be: How am I able to close such a PR if it already has been created?

**Reply by rarkins** [reply]
If you close the PR and delete the associated branch, it should not be recreated (but will remain visible in the dashboard). You can try that on your reproduction repo

**Reply by herndlm** [reply]
The version constraints in package.json would be interesting too, yeah. Could be something similar that I adapted, see previous comments / linked PR.

**Reply by karfau** [reply]
(Initially my reproduction was not minimal, since I had `automerge` enabled but of course no branch protection/checks enabled. Reverted that and tried again, works as expected.)

OK, closing those immortal PRs by directly deleting the branch in the list of branches works fine.
Thx for the hint, going to close some PRs.

Still the issues of the version not being in the PR title remains.

**Reply by herndlm** [reply]
If you look at https://github.com/renovatebot/renovate/pull/13830/files and the test - it could be that here `newValue` is 28.0 for one package and 28.1 for the other. That might be already enough to make it immortal. But I'm not sure, this is just a feeling.

This could even be considered expected behaviour I guess.

**Reply by karfau** [reply]
I can confirm what you are sying from the debug log of the reproduction.
But it looks as if the `newMajor` field has the information that is matching and should be enough in the case of a major upgrade.

```
DEBUG: packageFiles with updates
{
  "config": {
    "npm": [
      {
        "packageFile": "package.json",
        "deps": [
          {
            "depType": "devDependencies",
            "depName": "jest",
            "currentValue": "27.5.1",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "27.5.1",
            "depIndex": 0,
            "updates": [
              {
                "bucket": "major-28",
                "newVersion": "28.1.0",
                "newValue": "28.1.0",
                "releaseTimestamp": "2022-05-06T10:49:09.709Z",
                "newMajor": 28,
                "newMinor": 1,
                "updateType": "major",
                "branchName": "renovate/major-28-jest-monorepo"
              }
            ],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/facebook/jest",
            "sourceDirectory": "packages/jest",
            "homepage": "https://jestjs.io/",
            "currentVersion": "27.5.1",
            "isSingleVersion": true,
            "fixedVersion": "27.5.1"
          },
          {
            "depType": "devDependencies",
            "depName": "ts-jest",
            "currentValue": "27.1.4",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "27.1.4",
            "depIndex": 1,
            "updates": [
              {
                "bucket": "major-28",
                "newVersion": "28.0.2",
                "newValue": "28.0.2",
                "releaseTimestamp": "2022-05-08T07:47:18.625Z",
                "newMajor": 28,
                "newMinor": 0,
                "updateType": "major",
                "branchName": "renovate/major-28-jest-monorepo"
              }
            ],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/kulshekhar/ts-jest",
            "homepage": "https://kulshekhar.github.io/ts-jest",
            "currentVersion": "27.1.4",
            "isSingleVersion": true,
            "fixedVersion": "27.1.4"
          }
        ],
        "packageJsonName": "renovate-10710",
        "packageFileVersion": "1.0.0",
        "packageJsonType": "library",
        "npmLock": "package-lock.json",
        "managerData": {
          "yarnZeroInstall": false,
          "hasPackageManager": false
        },
        "skipInstalls": true,
        "constraints": {},
        "lockFiles": [
          "package-lock.json"
        ]
      }
    ]
  }
}
```

**Reply by rarkins** [reply]
Debugging existing code:

<img width="299" alt="image" src="https://user-images.githubusercontent.com/6311784/168457147-0583a1c0-5c6a-48a3-af4f-1249a812fec4.png">

<img width="439" alt="image" src="https://user-images.githubusercontent.com/6311784/168457154-3e580330-2451-483c-82a6-77c15544c2f4.png">

**Comment by rarkins** [comment, marked as answer]
PR: https://github.com/renovatebot/renovate/pull/15586
Example: https://github.com/renovate-reproductions/10710/pull/2
